RecoTracker/LSTCore/interface/alpaka/Constants.h

```diff
@@ -104,54 +104,80 @@ namespace SDL {
   constexpr float PT_CUT = 0.8f;
 #endif
 
-  const unsigned int MAX_BLOCKS = 80;
-  const unsigned int MAX_CONNECTED_MODULES = 40;
+  constexpr unsigned int MAX_BLOCKS = 80;
+  constexpr unsigned int MAX_CONNECTED_MODULES = 40;
 
-  const unsigned int N_MAX_PIXEL_SEGMENTS_PER_MODULE = 50000;
+  constexpr unsigned int N_MAX_PIXEL_SEGMENTS_PER_MODULE = 50000;
 
-  const unsigned int N_MAX_PIXEL_MD_PER_MODULES = 2 * N_MAX_PIXEL_SEGMENTS_PER_MODULE;
+  constexpr unsigned int N_MAX_PIXEL_MD_PER_MODULES = 2 * N_MAX_PIXEL_SEGMENTS_PER_MODULE;
 
-  const unsigned int N_MAX_PIXEL_TRIPLETS = 5000;
-  const unsigned int N_MAX_PIXEL_QUINTUPLETS = 15000;
+  constexpr unsigned int N_MAX_PIXEL_TRIPLETS = 5000;
+  constexpr unsigned int N_MAX_PIXEL_QUINTUPLETS = 15000;
 
-  const unsigned int N_MAX_PIXEL_TRACK_CANDIDATES = 30000;
-  const unsigned int N_MAX_NONPIXEL_TRACK_CANDIDATES = 1000;
+  constexpr unsigned int N_MAX_PIXEL_TRACK_CANDIDATES = 30000;
+  constexpr unsigned int N_MAX_NONPIXEL_TRACK_CANDIDATES = 1000;
 
-  const unsigned int size_superbins = 45000;
+  constexpr unsigned int size_superbins = 45000;
 
   //defining the constant host device variables right up here
-  ALPAKA_STATIC_ACC_MEM_GLOBAL const float miniMulsPtScaleBarrel[6] = {0.0052, 0.0038, 0.0034, 0.0034, 0.0032, 0.0034};
-  ALPAKA_STATIC_ACC_MEM_GLOBAL const float miniMulsPtScaleEndcap[5] = {0.006, 0.006, 0.006, 0.006, 0.006};
-  ALPAKA_STATIC_ACC_MEM_GLOBAL const float miniRminMeanBarrel[6] = {
+  namespace objLayers {
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr int kpLS =
```

Comment:
I'm not sure `ALPAKA_STATIC_ACC_MEM_GLOBAL` is needed with `constexpr`.
e.g. https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h#L34

Reply:
using a namespace to place the constants isn't quite clear to me. The constant inside a namespace should still have a clear meaning (after e.g. `using namespace`).
Perhaps the other way around it would make more sense `obj_pLS { int kLayers, kHits };` leaving space for other object specific constants. 
A struct/type would be better than a namespace (e.g.  for use in templates, for "traits" pattern)

Reply:
I agree with your point that a namespace like that could be misleading in some cases. Let's agree on a implementation then? I don't have a specific preference but maybe struct is the easiest? Let me know and I will apply the change (along with the rest of the fixes).

Reply:
how about `struct Params_pLS` and in corresponding implementation part may even simplify to `using Params = Params_pLS` and so on?
I'm not sure it's worth going to a template here with `kpLS, kLS` etc as an argument

Reply:
something that I didn't check though if it's practical to put all constants in the "global" Constants.h file.

Are these really used in multiple objects where the corresponding type will not be known (included)?
Can these params/constants be defined in `Segment.h` for pLS and LS?

Reply:
> Are these really used in multiple objects where the corresponding type will not be known (included)?

Yes. And multiple of them can be in other files, like `Kernels.h` as well.